forge-game/src/main/java/forge/game/staticability/StaticAbilityAlternativeCost.java

```diff
@@ -53,10 +54,11 @@ public static List<SpellAbility> alternativeCosts(final SpellAbility sa, final C
                     newSA.putParam("ManaRestriction", stAb.getParam("ManaRestriction"));
                 }
 
-                if (stAb.hasParam("AffectedZone")) {
-                    newSA.getRestrictions().setZone(ZoneType.smartValueOf(stAb.getParam("AffectedZone")));
-                } else if (!stAb.getHostCard().isImmutable() && stAb.hasParam("EffectZone") && !"All".equals(stAb.getParam("EffectZone"))) {
-                    newSA.getRestrictions().setZone(ZoneType.smartValueOf(stAb.getParam("EffectZone")));
+                if (!stAb.getHostCard().isImmutable()) {
```

Comment:
Hmn, this doesn't look like equivalent logic too me
why do you no longer want to transfer AffectedZone to the AltCostSA?

Reply:
I checked, there is no AlternativeCost effect with AffectedZone

Reply:
gotcha
can you just do `zones.get(0)` instead of `zones.stream().findFirst().get()` ?

Reply:
hmn I guess then it doesn't matter